src/processing.py

```diff
@@ -1,23 +1,20 @@
 import random
 import typing
-from string import ascii_lowercase
 
 from PIL import Image
 
+ASCII_VALUES = list(range(97, 123)) + [32, 33, 44, 46]  # "a"-"z", " ", "!", ",", "."
+
 
 def generate_alphabet(key: str) -> dict:
     """Generates the A-channel offset of each character in the alphabet using a pre-set key."""
     random.seed(key)
-    alphabet = {i: 0 for i in ascii_lowercase}
-    alphabet[" "] = 0
-    for i in alphabet.keys():
-        value = random.randint(1, 28)
-        if value in alphabet.values():
-            while True:
-                value = random.randint(1, 28)
-                if value not in alphabet.values():
-                    break
-        alphabet[i] = value
+    alphabet = {}
+    for i in ASCII_VALUES:
+        value = random.randint(0, len(ASCII_VALUES))
```

Comment:
Must be between 1 and max or 1 letter will be missed when reading the message from the image